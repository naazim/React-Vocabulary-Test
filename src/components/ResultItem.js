import React from 'react';

const ResultItem = (props) => {
    const {native, foreign, userAnswer} = props.word;
    return (
        <div className="result__item">
            <div className="result__icon">{props.index}</div>
            <div className="result__word">{native}</div>
            <div className="result__word">{foreign}</div>
            <div className="result__word">{userAnswer}</div>
            <div className={`result__icon ${foreign === userAnswer? 'result__icon--right' : 'result__icon--wrong'}`}>
                {foreign === userAnswer? '✔' : '✘'}
            </div>
        </div>
    );   
};

export default ResultItem;