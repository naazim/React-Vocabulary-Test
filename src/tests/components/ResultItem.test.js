import React from 'react';
import { shallow } from 'enzyme';
import ResultItem from '../../components/ResultItem';

describe('ResultItem', () => {
    let word = { native:"Hello", foreign:"Hallo", userAnswer: "Hal" };
    let wrapper = shallow(<ResultItem word={word}/>);

    test('should render ResultItem correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have an error indication for user\'s wrong answers',() =>{
        expect(wrapper.find('.result__error').length).toBe(1);
    });

    test('should have an error indication for user\'s right answers',() =>{
        word = { native:"Hello", foreign:"Hallo", userAnswer: "Hallo" };
        wrapper = shallow(<ResultItem word={word}/>);
        expect(wrapper.find('.result__error').length).toBe(0);
    });
});