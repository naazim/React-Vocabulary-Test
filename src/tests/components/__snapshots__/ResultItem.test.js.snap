// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem should render ResultItem correctly 1`] = `
<div
  className="result__item"
>
  <div
    className="result__count"
  >
    NaN
  </div>
  <div
    className="result__default"
  >
    <div
      className="result__word"
    >
      Hello
    </div>
    <div
      className="result__word"
    >
      Hallo
    </div>
  </div>
  <div
    className="result__answer result__error"
  >
    <span
      className="result__label"
    >
      Your answer:
    </span>
    Hal
  </div>
</div>
`;
